src/type/s2k/index.js

```diff
@@ -0,0 +1,35 @@
+import defaultConfig from '../../config';
+import Argon2S2k, { Argon2OutOfMemoryError } from './argon2';
+import OtherS2K from './generic';
+import enums from '../../enums';
+
+const allowedDefaultS2KTypes = new Set([enums.s2k.argon2, enums.s2k.iterated]);
```

Comment:
It's a bit nitpicky but this name is a bit strange to me - it's not about default S2K, but just the S2K types that are allowed to be generated. And, maybe instead of creating a set just for these two values, you could just have `if (config.s2kType !== enums.s2k.argon2 && config.s2kType !== enums.s2k.iterated)` in `newS2KFromConfig`?

Reply:
Yeah I will rename, but I'd leave the constant since it's a pattern we have in other places (e.g. `allowedUnhashedSubpackets` and `allowedPackets`) and I think it highlights the constraint without having to look much into the code ("when is `newPacketFromConfig` used" etc).